include/openenclave/edger8r/enclave.h

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/**
+ * @file enclave.h
+ *
+ * This file defines the internal functions and data-structures used by
+ * oeedger8r generated code on the enclave side.
+ * These internals are subject to change without notice.
+ *
+ */
+#ifndef _OE_EDGER8R_ENCLAVE_H
+#define _OE_EDGER8R_ENCLAVE_H
+
+#ifdef _OE_EDGER8R_HOST_H
+#error \
+    "edger8r/enclave.h and edger8r/host.h must not be included in the same compilation unit."
+#endif
+
+#include <openenclave/bits/defs.h>
+#include <openenclave/bits/result.h>
+#include <openenclave/bits/types.h>
+
+OE_EXTERNC_BEGIN
+
+/**
+ * The type of a function in ecall function table
+ */
+typedef void (*oe_ecall_func_t)(void*);
+
+/**
+ * Perform a high-level host function call (OCALL).
+ *
+ * Call the host function whose matching the given function_id.
+ * The host function is expected to have the following signature:
+ *
+ *     void (*)(void* args);
+ *
+ * Note that the return value of this function only indicates the success of
+ * the call and not of the underlying function. The OCALL implementation must
+ * define its own error reporting scheme via the arguments or return value.
+ *
+ * @param function_id The id of the host function that will be called.
+ * @param input_buffer Buffer containing inputs data.
+ * @param input_buffer_size Size of the input data buffer.
+ * @param output_buffer Buffer where the outputs of the host function are
+ * written to.
+ * @param output_buffer_size Size of the output buffer.
+ * @param output_bytes_written Number of bytes written in the output buffer.
+ *
+ * @return OE_OK the call was successful.
+ * @return OE_NOT_FOUND if the function_id does not correspond to a function.
+ * @return OE_INVALID_PARAMETER a parameter is invalid.
+ * @return OE_FAILURE the call failed.
+ * @return OE_BUFFER_TOO_SMALL the input or output buffer was smaller than
+ * expected.
+ */
+oe_result_t oe_call_host_function(
+    size_t function_id,
+    void* input_buffer,
+    size_t input_buffer_size,
+    void* output_buffer,
+    size_t output_buffer_size,
+    size_t* output_bytes_written);
+
+/**
+ * Check whether the given buffer is strictly within the enclave.
+ *
+ * Check whether the buffer given by the **ptr** and **size** parameters is
+ * strictly within the enclave's memory. If so, return true. If any
+ * portion of the buffer lies outside the enclave's memory, return false.
+ *
+ * @param ptr The pointer pointer to buffer.
+ * @param size The size of buffer
+ *
+ * @retval true The buffer is strictly within the enclave.
+ * @retval false At least some part of the buffer is outside the enclave, or
+ * the arguments are invalid. For example, if **ptr** is null or **size**
+ * causes arithmetic operations to wrap.
+ *
+ */
+bool oe_is_within_enclave(const void* ptr, size_t size);
```

Comment:
we don't already have these functions in the main SDK?

Reply:
Ok. I will remove them for now. I suspect this is where these functions will eventually end up.